vm/ubpf_vm.c

```diff
@@ -298,13 +299,14 @@ ubpf_exec(const struct ubpf_vm* vm, void* mem, size_t mem_len, uint64_t* bpf_ret
 
     stack = calloc(UBPF_STACK_SIZE, 1);
     if (!stack) {
-        return -1;
+        return_value = -1;
+        goto exit;
     }
 
     stack_frames = calloc(UBPF_MAX_CALL_DEPTH, sizeof(struct ubpf_stack_frame));
     if (!stack_frames) {
-        free(stack);
-        return -1;
+        return_value = -1;
+        goto exit;
```

Comment:
Here I think that our cleanup path will need to free `stack` because we have already allocated it by the time that we got here.

Reply:
Not sure I follow. Cleanup always frees stack_frames and stack, so doing it here seems redundant.

Reply:
The confusion was simply as a result of the typo between `exit` and `cleanup`. All better now.